core/primitives/src/block.rs

```diff
@@ -284,6 +284,7 @@ impl Block {
     pub fn produce(
         this_epoch_protocol_version: ProtocolVersion,
         next_epoch_protocol_version: ProtocolVersion,
+        latest_protocol_version: ProtocolVersion,
```

Comment:
nit: It's a bit of an anti pattern to have multiple arguments of the same type, it's all too easy to mistake the order. I don't really have a better suggestion though. 

nit2: Is it still latest_protocol_version or "voting_protocol_version" (with name consistent with the rest of the PR).